src/api-service/__app__/onefuzzlib/request.py

```diff
@@ -6,36 +6,81 @@
 import json
 import logging
 import os
+import urllib
 from typing import TYPE_CHECKING, Optional, Sequence, Type, TypeVar, Union
 from uuid import UUID
 
 from azure.functions import HttpRequest, HttpResponse
+from memoization import cached
 from onefuzztypes.enums import ErrorCode
 from onefuzztypes.models import Error
 from onefuzztypes.responses import BaseResponse
+from pydantic import BaseModel  # noqa: F401
 from pydantic import ValidationError
 
+from .azure.group_membership import create_group_membership_checker
+from .config import InstanceConfig
 from .orm import ModelMixin
+from .request_access import RequestAccess
 
 # We don't actually use these types at runtime at this time.  Rather,
 # these are used in a bound TypeVar.  MyPy suggests to only import these
 # types during type checking.
 if TYPE_CHECKING:
     from onefuzztypes.requests import BaseRequest  # noqa: F401
-    from pydantic import BaseModel  # noqa: F401
+
+
+def get_rules() -> Optional[RequestAccess]:
+    config = InstanceConfig.fetch()
+    if config.api_access_rules:
+        return RequestAccess.build(config.api_access_rules)
+    else:
+        return None
+
+
+@cached(ttl=60)
+def get_rules_cached() -> Optional[RequestAccess]:
+    return get_rules()
 
 
 def check_access(req: HttpRequest) -> Optional[Error]:
-    if "ONEFUZZ_AAD_GROUP_ID" in os.environ:
-        message = "ONEFUZZ_AAD_GROUP_ID configuration not supported"
-        logging.error(message)
-        return Error(
-            code=ErrorCode.INVALID_CONFIGURATION,
-            errors=[message],
-        )
+    if "NO_REQUEST_ACCESS_RULES_CACHE" in os.environ:
+        rules = get_rules()
     else:
+        rules = get_rules_cached()
+
+    if not rules:
```

Comment:
The check is that if member id is in rule.allowed_group_ids -> then they are allowed access

If there are no rules, doesn't it mean that no-one should be able to have access, since there are zero allowed_group_ids ?

Reply:
Nope, No rules means there are no restriction on the API.

Reply:
So we are not validating if user is allowed to access, we are validating whether user is NOT allowed ?

Reply:
Because down below there is check for "is_member" to mean that user has access and that seems like a contradiciton

Reply:
Discussed offline